code/modules/surgery/bodyparts/helpers.dm

```diff
@@ -129,18 +129,20 @@
 
 ///Remove a specific embedded item from the carbon mob
 /mob/living/carbon/proc/remove_embedded_object(obj/item/embedded)
-	SEND_SIGNAL(src, COMSIG_CARBON_EMBED_REMOVAL, embedded)
+	if (!embedded.get_embed()?.owner == src)
+		return
+	embedded.embed_data.remove_embedding()
```

Comment:
`get_embed()`?

Reply:
actually you're right i should probably convert embed_data to helpers